tests/conftest.py

```diff
@@ -214,20 +214,34 @@ def pytest_addoption(parser):
                      help="Delay between qutebrowser commands.")
     parser.addoption('--qute-profile-subprocs', action='store_true',
                      default=False, help="Run cProfile for subprocesses.")
-    parser.addoption('--qute-bdd-webengine', action='store_true',
-                     help='Use QtWebEngine for BDD tests')
+    parser.addoption('--qute-bdd-backend', action='store',
+                     choices=['webkit', 'webengine'], help='Set backend for BDD tests')
 
 
 def pytest_configure(config):
-    webengine_arg = config.getoption('--qute-bdd-webengine')
-    webengine_env = os.environ.get('QUTE_BDD_WEBENGINE', 'false')
-    config.webengine = webengine_arg or webengine_env == 'true'
-    # Fail early if QtWebEngine is not available
-    if config.webengine:
+    try:
+        # Try to use QtWebkit as the default backend
+        import PyQt5.QtWebKitWidgets
+        config.backend = 'webkit'
+    except ImportError:
+        # Try to use QtWebEngine as a fallback and fail early
+        # if that's also not available
         import PyQt5.QtWebEngineWidgets
+        config.backend = 'webengine'
+
+    webengine_arg = config.getoption('--qute-bdd-backend')
+    webengine_env = os.environ.get('QUTE_BDD_BACKEND')
+    config.webengine = (webengine_arg == 'webengine' or
+                        webengine_env == 'webengine' or
+                        config.backend == 'webengine')
+
     earlyinit.configure_pyqt()
 
 
+def pytest_report_header(config):
+    return f'backend: {config.backend}'
```

Comment:
I'm wondering if we should report the backend's version additionally to its name. Something like:
```suggestion
    if backend == 'webkit':
        version = version.qWebKitVersion
    elif backend == 'webengine':
        version = version.PYQT_WEBENGINE_VERSION_STR
    else:
        raise NotImplementedError(f"unsupported backend: {backend}")

    return f'backend: {config.backend} ({version})'
```

Reply:
Sounds like a good idea! Might want to use `str(version.qtwebengine_versions(avoid_init=True))` for the QtWebEngine case to get some more information. Also, note that `qWebKitVersion` is a function, so you'll need to call it. Finally, please use `utils.Unreachable(backend)` instead of `NotImplementedError` (which is typically used for abstract methods).

Reply:
Thanks for the very useful hints! I will implement that later today.